sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableReadSchemaTransformProvider.java

```diff
@@ -100,60 +100,55 @@ public List<String> outputCollectionNames() {
     return Collections.singletonList(OUTPUT_TAG);
   }
 
-  /** Configuration for reading from BigTable. */
+  /** Configuration for reading from Bigtable. */
   @DefaultSchema(AutoValueSchema.class)
   @AutoValue
-  public abstract static class BigTableReadSchemaTransformConfiguration {
-    /** Instantiates a {@link BigTableReadSchemaTransformConfiguration.Builder} instance. */
+  public abstract static class BigtableReadSchemaTransformConfiguration {
+    /** Instantiates a {@link BigtableReadSchemaTransformConfiguration.Builder} instance. */
+    public void validate() {
+      String emptyStringMessage =
+          "Invalid Bigtable Read configuration: %s should not be a non-empty String";
+      checkArgument(!this.getTableId().isEmpty(), String.format(emptyStringMessage, "table"));
+      checkArgument(!this.getInstanceId().isEmpty(), String.format(emptyStringMessage, "instance"));
+      checkArgument(!this.getProjectId().isEmpty(), String.format(emptyStringMessage, "project"));
+    }
+
     public static Builder builder() {
-      return new AutoValue_BigTableReadSchemaTransformProvider_BigTableReadSchemaTransformConfiguration
+      return new AutoValue_BigtableReadSchemaTransformProvider_BigtableReadSchemaTransformConfiguration
           .Builder();
     }
 
-    public abstract String getTable();
+    public abstract String getTableId();
 
-    public abstract String getInstance();
+    public abstract String getInstanceId();
 
-    public abstract String getProject();
+    public abstract String getProjectId();
 
-    /** Builder for the {@link BigTableReadSchemaTransformConfiguration}. */
+    /** Builder for the {@link BigtableReadSchemaTransformConfiguration}. */
     @AutoValue.Builder
     public abstract static class Builder {
-      public abstract Builder setTable(String table);
-
-      public abstract Builder setInstance(String instance);
-
-      public abstract Builder setProject(String project);
-
-      abstract BigTableReadSchemaTransformConfiguration autoBuild();
+      public abstract Builder setTableId(String table);
```

Comment:
Nit: change variable names to match the setters (tableId, instanceId, projectId).